cli.c

```diff
@@ -116,6 +121,118 @@ NRF_CLI_CREATE_STATIC_SUBCMD_SET(m_sub_cam){
     NRF_CLI_SUBCMD_SET_END};
 NRF_CLI_CMD_REGISTER(cam, &m_sub_cam, "camera", cmd_cam);
 
+
+/******************************************************************************/
+/*!                SPI Commands                                               */
+static void cmd_spi_readChipID(nrf_cli_t const *p_cli, size_t argc, char **argv)
+{
+  // read chip id = 0x24 at address 0x00
+  int8_t id = 0x00;
+  //uint8_t id = imuRead(0x00);
+
+  // log the data returned
+  NRF_LOG_RAW_INFO("%08d [spi IMU] chip_id = %x\n", systemTimeGetMs(), id);
+}
+
+// Read specific register
+static void cmd_spi_readReg(nrf_cli_t const *p_cli, size_t argc, char **argv)
+{
+  uint8_t regAddr = (uint8_t)strtol(argv[1], NULL, 16);
+
+  //uint8_t regVal = imuRead(regAddr);
+  //NRF_LOG_RAW_INFO("reg %x: %x\n", regAddr, regVal);
+
+}
+
+// Set specific register
+static void cmd_spi_setReg(nrf_cli_t const *p_cli, size_t argc, char **argv)
+{
+  uint8_t regAddr = (uint8_t)strtol(argv[1], NULL, 16);
+  uint8_t regVal = (uint8_t)strtol(argv[2], NULL, 16);
+
+  NRF_LOG_RAW_INFO("reg %x: %x\n", regAddr, regVal);
+
+  //imuWrite(regAddr, regVal);
+
+}
+
+static void cmd_spi(nrf_cli_t const *p_cli, size_t argc, char **argv)
+{
+  ASSERT(p_cli);
+  ASSERT(p_cli->p_ctx && p_cli->p_iface && p_cli->p_name);
+
+  if ((argc == 1) || nrf_cli_help_requested(p_cli))
+  {
+    nrf_cli_help_print(p_cli, NULL, 0);
+    return;
+  }
+
+  if (argc != 2)
+  {
+    nrf_cli_fprintf(p_cli, NRF_CLI_ERROR, "%s: bad parameter count\n", argv[0]);
+    return;
+  }
+
+  nrf_cli_fprintf(p_cli, NRF_CLI_ERROR, "%s: unknown parameter: %s\n", argv[0], argv[1]);
+}
+
+NRF_CLI_CREATE_STATIC_SUBCMD_SET(m_sub_spi)
+{
+    NRF_CLI_CMD(readReg, NULL, "spiBus, data, length", cmd_spi_readReg),
```

Comment:
the code for these commands looks commented out, if we're not using them remove. Or create an imu transfer command which lets us write directly.